src/br_location_class.py

```
@@ -149,6 +163,67 @@ def get_state_and_city(self) -> tuple[str, str, str]:
+        # Stage 2: count per-state, batch-draw cities
+        state_name_counts = Counter(drawn_state_names)
+        city_draws: dict[str, list[str]] = {}
+
+        for state_name, count in state_name_counts.items():
+            abbr = state_abbr_map[state_name]
+            if abbr in self.city_names_by_state:
+                city_draws[state_name] = random.choices(
+                    self.city_names_by_state[abbr],
+                    weights=self.city_weights_by_state[abbr],
+                    k=count,
+                )
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)